packages/dd-trace/test/debugger/devtools_client/condition.spec.js

```diff
@@ -0,0 +1,261 @@
+'use strict'
+
+const { inspect } = require('util')
+require('../../setup/mocha')
+
+const compile = require('../../../src/debugger/devtools_client/condition')
+
+class CustomObject {}
+
+const weakKey = { weak: 'key' }
+
+const testCases = [
+  // Plain references
+  [{ ref: 'foo' }, { foo: 42 }, 42],
+  [{ ref: 'foo' }, {}, new ReferenceError('foo is not defined')], // TODO: Will this actually throw in CDP?
+  [{ getmember: [{ ref: 'obj' }, 'foo'] }, { obj: { foo: 'test-me' } }, 'test-me'],
+  [
+    { getmember: [{ getmember: [{ ref: 'obj' }, 'foo'] }, 'bar'] },
+    { obj: { foo: { bar: 'test-me' } } },
+    'test-me'
+  ],
+  [
+    { getmember: [{ ref: 'set' }, 'foo'] },
+    { set: new Set(['foo', 'bar']) },
+    new Error('Aborting because accessing a Set or WeakSet is not allowed')
+  ],
+  [
+    { getmember: [{ ref: 'wset' }, { ref: 'key' }] },
+    { key: weakKey, wset: new WeakSet([weakKey]) },
+    new Error('Aborting because accessing a Set or WeakSet is not allowed')
+  ],
+  [
+    { getmember: [{ ref: 'map' }, 'foo'] },
+    { map: new Map([['foo', 'bar']]) },
+    new Error('Aborting because accessing a Map or WeakMap is not allowed')
+  ],
+  [
+    { getmember: [{ ref: 'wmap' }, { ref: 'key' }] },
+    { key: weakKey, wmap: new WeakMap([[weakKey, 'bar']]) },
+    new Error('Aborting because accessing a Map or WeakMap is not allowed')
+  ],
+  [
+    { getmember: [{ ref: 'obj' }, 'getter'] },
+    { obj: Object.create(Object.prototype, { getter: { get () { return 'x' } } }) },
+    new Error('Aborting because of possible side effects')
+  ],
+
+  // References with operations
+  [{ len: { ref: 'foo' } }, { foo: 'hello' }, 5],
```

Comment:
lets add test for 'len' 'size' with getters defined

Reply:
Done 👍